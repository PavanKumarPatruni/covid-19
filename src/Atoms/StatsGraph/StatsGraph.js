import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { Doughnut } from 'react-chartjs-2';

import { Colors } from '../../utils/contants';

const propTypes = {
  stats: PropTypes.object,
};

const StatsGraph = ({ stats }) => {
  const [statsData, setStatsData] = useState({});

  const getData = (item, key) => {
    return item && item[key];
  };

  useEffect(() => {
    if (stats) {
      const active = getData(stats, 'active');
      const recovered = getData(stats, 'recovered');
      const deaths = getData(stats, 'deaths');

      const data = {
        labels: ['Active', 'Recovered', 'Deceased'],
        datasets: [
          {
            backgroundColor: [Colors.RED, Colors.GREEN, Colors.GREY],
            hoverBackgroundColor: [Colors.RED, Colors.GREEN, Colors.GREY],
            borderColor: Colors.PRIMARY,
            data: [active, recovered, deaths],
          },
        ],
      };

      setStatsData(data);
    }
  }, [stats]);

  return (
    <div className="flex flex-justify-center mt4">
      <Doughnut
        width={400}
        height={400}
        data={statsData}
        legend={{ display: true }}
      />
    </div>
  );
};

StatsGraph.defaultProps = {
  stats: {},
};

StatsGraph.propTypes = propTypes;

export default StatsGraph;
