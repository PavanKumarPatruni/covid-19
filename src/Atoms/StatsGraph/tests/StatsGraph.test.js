import React from 'react';
import { shallow, mount } from 'enzyme';

import StatsGraph from '../StatsGraph';

describe('StatsGraph component', () => {
  const props = {
    stats: {
      active: '44351',
      confirmed: '67700',
      deaths: '2215',
      deltaconfirmed: '524',
      deltadeaths: '2',
      deltarecovered: '160',
      lastupdatedtime: '11/05/2020 15:41:22',
      recovered: '21130',
      state: 'Total',
      statecode: 'TT',
      statenotes: '',
    },
  };

  test('match snapshot correctly', () => {
    const wrapper = shallow(<StatsGraph {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('match snapshot correctly', () => {
    const wrapper = mount(<StatsGraph {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
