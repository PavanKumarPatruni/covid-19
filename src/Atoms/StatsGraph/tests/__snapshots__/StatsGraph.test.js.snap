// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsGraph component match snapshot correctly 1`] = `
<div
  className="flex flex-justify-center mt4"
>
  <Doughnut
    data={Object {}}
    height={400}
    legend={
      Object {
        "display": true,
      }
    }
    width={400}
  />
</div>
`;
